pandas/io/parsers/readers.py

```diff
@@ -1359,9 +1359,10 @@ def read_fwf(
 
 class TextFileReader(abc.Iterator):
     """
+    Passed dialect overrides any of the related parser options.
 
-    Passed dialect overrides any of the related parser options
-
+    Only __enter__, __exit__ and __next__ are public. All other
+    attributes are considered private and can change.
```

Comment:
This is probably quite cryptic if the idea here is to make this documentation public. I would have something like:
```
Iterator class to process text files in chunks.

An instance of this class is returned by `read_csv` when it is processed in
chunks, instead of returning a single `DataFrame`.

When iterating over `TestFileReader`, every item returned will be a dataframe.

Examples
---------
>>> with pandas.read_csv(..., iterator=True) as text_file_reader:
...     for df in text_file_reader:
...         ...
```